### DISCUSSION TITLE
VQE performance on the CUDA-Q

### DISCUSSION THREAD

**Discussion by amihashemi** [NVIDIA/cuda-quantum#1684] [Q&A]
Hi,
I'm wondering if there has been any study on comparing VQE on hybrid CUDA-Q compared with the current quantum computers or classical solvers, I would appreciate if anyone can help. I see that there is a recent article on VQE (https://developer.nvidia.com/blog/new-nvidia-cuda-q-features-boost-quantum-application-performance/), but it seems to only include different GPU simulations, no comparison with other computational unit...

Thanks
Ami

**Comment by Advait251206** [comment]
This is a very good question—and the short answer is:

> There is **no broadly accepted, apples-to-apples benchmark study** showing CUDA-Q VQE consistently outperforming real quantum hardware or classical VQE solvers across problem classes.

But there *is* meaningful context on where CUDA-Q sits in the landscape.

---

# 1. First: what CUDA-Q VQE is actually optimized for

CUDA-Q VQE (hybrid workflow) is primarily designed to accelerate:

- statevector / tensor network simulation of circuits
- parameter sweep evaluation (expectation values)
- gradient evaluation loops
- batched circuit execution on GPUs

So it improves:

> ✔ classical simulation speed of VQE  
> ❌ not quantum advantage vs hardware

---

# 2. Why “comparison with quantum computers” is tricky

Comparing CUDA-Q VQE vs real QPUs is not straightforward because:

### Different objectives:
| Platform | What it measures |
|----------|------------------|
| CUDA-Q | classical simulation efficiency |
| QPU | physical quantum evolution |
| VQE experiment | hardware noise + sampling |

So CUDA-Q is:

> a **classical emulator of VQE**, not a competing quantum device

---

# 3. CUDA-Q vs classical solvers (important nuance)

There *are* partial comparisons in NVIDIA materials and community experiments:

## CUDA-Q tends to outperform when:

### ✔ large statevector workloads
- 20–40+ qubits (depending on backend)
- deep ansatz circuits
- GPU-accelerated expectation evaluations

### ✔ repeated Hamiltonian expectation evaluation
- many parameter updates
- batched circuit execution

---

## Classical solvers still win when:

### ✔ small / structured problems
- tensor network contraction (ITensor / Quimb)
- exact diagonalization (small systems)
- strongly structured Hamiltonians

---

# 4. Why CUDA-Q VQE is fast (core reason)

CUDA-Q gains come from:

### GPU acceleration of:
- full statevector evolution
- batched parameter evaluation
- parallel expectation computations

So instead of:

```
CPU loop over circuits
```

you get:

```
GPU-parallelized circuit evolution
```

---

# 5. Why there is no “standard benchmark paper”

There are three reasons:

---

## (A) VQE is not a single algorithm

It depends heavily on:

- ansatz choice (hardware efficient, UCCSD, etc.)
- optimizer (COBYLA, SPSA, gradient-based)
- Hamiltonian structure

So results vary wildly.

---

## (B) CUDA-Q evolves quickly

Benchmarks become outdated fast due to:

- SVS improvements
- TensorNet optimizations
- multi-GPU scaling changes

---

## (C) QPU performance is not stable

Real hardware varies by:

- noise model
- calibration drift
- backend provider (IBM, IonQ, Rigetti)

So comparisons are not reproducible.

---

# 6. What *does exist* in practice

From NVIDIA CUDA-Q ecosystem + papers/blogs:

### ✔ GPU vs CPU VQE comparisons:
- often show **10×–100× speedups** in expectation evaluation loops
- especially for large statevector simulations

### ✔ Hybrid workflows:
- CUDA-Q used as **classical surrogate for QPU execution**
- useful for algorithm prototyping and optimization research

But:

> these do NOT claim quantum advantage or QPU superiority

---

# 7. Key conceptual takeaway

Think of CUDA-Q VQE as:

> a **high-performance classical simulator for variational quantum algorithms**

not as:

> a competitor to quantum hardware

---

# 8. When CUDA-Q VQE is most useful

✔ Prototyping ansatz circuits  
✔ Benchmarking optimization strategies  
✔ Large-scale classical emulation  
✔ GPU-accelerated Hamiltonian evaluation  

---

# 9. Direct answer

- ❌ No definitive study shows CUDA-Q VQE beating quantum computers (not meaningful comparison)
- ❌ No universal benchmark vs all classical solvers
- ✔ CUDA-Q significantly accelerates **classical VQE simulation on GPUs**
- ✔ Best gains appear in large, repeated expectation-value workloads

---

# If you want

I can also break down:

- when CUDA-Q VQE beats QuTiP / PennyLane / Qiskit Aer
- how scaling behaves with qubit count + ansatz depth
- or what “realistic quantum advantage benchmarks” look like today

---

If this answer helped or pointed you in the right direction, I'd appreciate it if you could mark it as the accepted answer so it's easier for others with the same issue to find.

Also, if you found my contribution useful, I'd appreciate it if you could check out my GitHub profile, follow me, and star any repositories you find interesting.

GitHub: https://github.com/Advait251206
